src/Xrefcheck/Util.hs

```diff
@@ -41,3 +46,54 @@ infixr 0 -:
 -- | Options that we use to derive JSON instances for config types.
 aesonConfigOption :: Aeson.Options
 aesonConfigOption = aesonPrefix camelCase
+
+invalidURIVerbose :: Char -> Int -> Text -> Text
+invalidURIVerbose characterInQuestion position link = fmt . indentF 3 . unlinesF $
+  pinpointInvalidCharacter <>
+  suggestPercentEncoding
+  where
+    pinpointInvalidCharacter  =
+      [ "unexpected character " <> show characterInQuestion
+      , style Bold (color Magenta "|")
+      , style Bold (color Magenta "|") <> "  " <>
+        applyPrettyAt position link Bold Magenta
+      , style Bold (color Magenta "|") <> "  " <>
+        T.replicate position " " <> style Bold (color Magenta "^")
+      , show characterInQuestion <>
+        " is not allowed to be explcitily used in the URIs by the WHATWG URL standard"
+      , "Consider using its percent-encoded counterpart: " <>
+        style Bold (color Magenta $ octet characterInQuestion)
+      ]
+
+    suggestPercentEncoding = maybeToList $ do
+      paired <- characterInQuestion `M.lookup` pairedEnclosingGlyphs
```

Comment:
In `illegalCharactersToDisplay` (in the `Verify` module) we're saying that the only characters we consider "illegal" and deserve special handling are `[` and `]`. But here we're saying `[ ] { } < >` are also illegal?